src/test/groovy/org/prebid/server/functional/tests/privacy/GppAuctionSpec.groovy

```diff
@@ -0,0 +1,180 @@
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 1
+    }
+
+    def "PBS should populate gdpr to 0 when regs.gdpr is not specified and gppSid not contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = null
+            regs.gppSid = [PBSUtils.getRandomNumberWithExcept(2)]
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 0
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid contains 2 and regs.gdpr isn't 1"() {
+        given: "Default bid request with gppSid and gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = 0
+            regs.gppSid = [2]
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid not contains 2 and regs.gdpr isn't 0"() {
+        given: "Default bid request with gppSid and gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = 1
+            regs.gppSid = [PBSUtils.getRandomNumberWithExcept(2)]
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+    }
+
+    def "PBS should emit warning when GPP string invalid"() {
+        given: "Default bid request with invalid gpp"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gpp = "Invalid_GPP_Consent_String"
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP string invalid: Undecodable FibonacciIntegerRange '101111'"]
```

Comment:
PBS should return more adequate error message. `Undecodable FibonacciIntegerRange '101111'` means nothing to the end user.